### DISCUSSION TITLE
ERROR!!!: Too old Airflow version !

### DISCUSSION THREAD

**Discussion by AbrahanTorres** [apache/airflow#31521] [General]
### Apache Airflow version

2.6.1

### What happened

Hi team.

Please help me with an issue.

I have got an issue recently, it is new, weeks ago It did not happened. 

I use in the dockerfile FROM apache/airflow:latest
![image](https://github.com/apache/airflow/assets/102258512/07f0cbd3-3b9e-4ea7-b86f-a86ef79812e9)

And in the docker-compose.yml also the latest version

![image](https://github.com/apache/airflow/assets/102258512/9df57625-29ad-4d84-913e-c3702d14658e)

I executed the docker command docker build and everthing is OK, The problem happend when I run the docker-compose up command.

![image](https://github.com/apache/airflow/assets/102258512/4e11f2b7-c70b-45f5-aef5-600072f34275)

How can I fixed this?

### What you think should happen instead

It sholud works as before.

### How to reproduce

    1. DOCKERFILE
    docker build . --tag extending_airflow:latest

    2. DOCKER-COMPOSE 
    docker-compose up

### Operating System

windows

### Versions of Apache Airflow Providers

latest

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
Likely some of your requirements is bringing airflow down - See this chapter and note how you can detect this and find out which requirement does it:  https://airflow.apache.org/docs/docker-stack/build.html#example-of-adding-packages-from-requirements-txt

**Reply by AbrahanTorres** [reply]
From requirements.txt it was setted to install a specific old version of airflow, I fixed it.

Also there was necessary to delete log directory before executed docker-compose.yml.

Now works, thank you.

**Comment by potiuk** [comment]
Converted to a discussion if needed.

**Comment by kcphila** [comment]
I ran into the same issue when building against 2.8.3.  The default image does not have `gosu` installed, which is called in the bootstrapping process to determine the version of airflow.  So I had to add a `RUN apt-get install -y gosu` to the dockerfile. That might be something that can/should be included in the source image.

**Reply by potiuk** [reply]
FYI Gosu has been removed due to security vulnerabilities it caused. I think it has been missing in changelog but I will update it

**Reply by potiuk** [reply]
It never served any purpose it was just **there** accidentally - and yes - if you want to get it, you need to install it.

**Reply by potiuk** [reply]
> which is called in the bootstrapping process to determine the version of airflow.

BTW. What kind of bootstrapping requires `gosu` to check airflow version? It puzzled me @kcphila - can you please explain ?

**Reply by kcphila** [reply]
Clearly it's being used somewhere. Here's the output that keyed me into the issue.  

```
5432"
postgres-1           | 2024-03-12 17:47:25.594 UTC [27] LOG:  database system was shut down at 2024-03-12 17:38:16 UTC
postgres-1           | 2024-03-12 17:47:25.600 UTC [1] LOG:  database system is ready to accept connections
airflow-init-1       | /bin/bash: line 4: gosu: command not found
airflow-init-1       |
airflow-init-1       | ERROR!!!: Too old Airflow version !
airflow-init-1       | The minimum Airflow version supported: 2.2.0. Only use this or higher!
```

This was more or less or the docker file

```
FROM apache/airflow:2.8.3-python3.11

USER airflow

WORKDIR /opt/airflow

RUN mkdir /tmp/requirements
COPY requirements.txt /tmp/requirements/requirements.txt

RUN python -m pip install "apache-airflow[postgres]==2.8.3" --constraint "https://raw.githubusercontent.com/apache/airflow/constraints-2.8.3/constraints-3.8.txt"

RUN python -m pip install --no-cache-dir --upgrade -r /tmp/requirements/requirements.txt

RUN rm -rf /tmp/requirements

```

**Reply by kcphila** [reply]
On the off hand that it might be related, I also inadvertently put `jwt` in the requirements file when I meant `PyJWT` - and Airflow also depends on `PyJWT` and this broke the init.  the jwt error didn't occur until after I added the gosu install to the dockerfile

**Reply by potiuk** [reply]
Well. It's likely used by some docker compose of yours or chart - but I can't find any use of it it Airflow - can you check WHERE it is coming from - I am curious :)

**Reply by kcphila** [reply]
I'm only using the vanilla docker compose .... wait..... aha! Right, I'm using a docker-compose yaml from Airflow 2.7! I forgot to update that.  Looks like that explains it.

**Reply by potiuk** [reply]
Ahh... Interesting - can't recall 2.7 having gosu, but maybe :)

**Reply by Taragolis** [reply]
Version checks was exists in docker compose sample until 2.8, which use gosu. I guess this was the only place where gosu uses

**Reply by potiuk** [reply]
yeah. It's evil to keep it.

**Reply by kcphila** [reply]
Yeah - I checked the docker file, it was from 2.7.3 

I guess the takeaway for anyone in the future who comes across this error:

- check to make sure your docker-compose.yaml file is for the same airflow version as you are running
- If the error message has `jwt` somewhere in it, make sure you didn't install `jwt` or `python-jwt` as they will conflict with `PyJWT` during init
- Otherwise, if you built a custom docker image, it might be something there - and if not, it may be in some changes you made to the docker-compose.yaml

**Reply by Taragolis** [reply]
I've found my old comment: https://github.com/apache/airflow/pull/35513#discussion_r1385481430

> I guess the takeaway for anyone in the future who comes across this error:

Some additional points:

- Avoid void to use `_PIP_ADDITIONAL_REQUIREMENTS`
- Avoid to use sample docker-compose.yaml in production

**Reply by kcphila** [reply]
Well, you shouldn't be using docker in production in general :)

**Reply by potiuk** [reply]
Or computers, in general :)
